doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -7,14 +7,62 @@ Security
    :local:
    :depth: 3
 
-Nordic Matter samples leverage security features supported in the |NCS| that can be divided into three major categories:
+Nordic Matter samples leverage security features supported in the |NCS| that can be divided into four major categories:
 
+* Trusted Execution Environment
 * Cryptography
 * Trusted storage
 * Securing production devices
 
 In the following sections you will learn more details about each listed category.
 
+Trusted Execution Environment
+*****************************
+
+In Matter samples, you can use two types of Trusted Execution Environments (TEE) to provide a secure environment for the execution of security-sensitive code, depending on the SoC:
+
+* nRF54L Series devices can use Trusted Firmware-M (TF-M).
+* nRF54H Series devices can use the Secure Domain.
+
+All cryptographic operations are performed using the PSA Cryptography API.
+
+nRF54H Secure Domain
+====================
+
+The Secure Domain is a dedicated domain that executes a pre-compiled, Nordic Semiconductor-signed firmware component.
+This firmware allows the execution of all secure operations in a separate domain, providing a secure environment for security-sensitive Matter code.
+
+In Matter samples, we use security services exposed by the Secure Domain to perform all cryptographic operations during commissioning to Matter and while operating in the Matter network.
+All keys used by Matter, such as session keys, DAC private keys, and other keys, are stored in the Secure Domain's secure space and are isolated from local domains.
+
+To learn more about the secure domain, see the :ref:`ug_nrf54h20_architecture_cpu` user guide.
+
+nRF54L Trusted Firmware-M (TF-M)
+================================
+
+Matter samples based on the nRF54L SoC support the Trusted Firmware-M (TF-M) library.
+All cryptographic operations within the Matter stack are performed by utilizing the PSA Cryptography API and executed in the secure TF-M environment.
+Application and Matter codes are executed in the Non-Secure Processing Environment, whereas MCUboot and TF-M are executed in the Secure Processing Environment (see :ref:`app_boards_spe_nspe` for more information).
+This isolation of firmware parts provides security for the execution of security-sensitive code.
+The secure materials like Matter Session keys, DAC private key and other keys, are stored in the TF-M secure storage utilizing the :ref:`tfm_encrypted_its` module.
+Matter samples utilize the full TF-M library, so you cannot use the :ref:`tfm_minimal_build` version of TF-M.
```

Comment:
Same comment as above -- link + include info in the bullet point entry? Why duplicate sections when we can link?